elastic/apm-server
# Various integration package enhancements and fixes

## Motivation/summary

- Add `client.geo` to internal_metrics, so RUM breakdown metrics can be indexed. The internal metrics data stream is strictly
  mapped.
- Remove our README generator, update to use the elastic-package generator instead. This means we no longer have ECS badges; we should enhance the elastic-package generator later if this is important. I have also removed the sample documents, as we have these in the elastic.co APM documentation.
- Stop generating integration package fields and ingest pipelines. Prune unused fields per data stream, and introduce data stream specific ingest pipelines. This means that any changes to `_meta/fields.yml` and `ingest/pipeline/definition.yml` that should be reflected in the integration package need be made manually from now on.
- Use `constant_keyword` for `processor.name` and `processor.event` fields where we can.
- Use `external: ecs` for ECS fields, so we get the canonical ECS field descriptions and types.

Now that we're using `external: ecs` in fields/ecs.yml, we need to use the output of `elastic-package build` rather than using the integration package source directly. This goes for the system tests (i.e. what we mount into the package-registry container), and also for what we copy to package-storage.

## Checklist

~- [ ] Documentation has been updated~

## How to test these changes

1. Install integration
2. Ingest RUM breakdown metrics, check they are indexed with `client.geo.location`
3. Manually inspect installed ingest pipelines for each data stream, checking that they look correct

## Related issues

Closes https://github.com/elastic/apm-server/issues/6344
Closes https://github.com/elastic/apm-server/issues/6001
Closes https://github.com/elastic/apm-server/issues/5602
Closes https://github.com/elastic/apm-server/issues/3871
